SchedulerBot/SchedulerBot.Business.Entities/ExecutionResult/ExecutionResult.cs

```diff
@@ -0,0 +1,208 @@
+		/// </summary>
+		/// <param name="errorCode">The error code.</param>
+		/// <param name="errorErrorMessage">The error message.</param>
+		protected ExecutionResult(ExecutionErrorCode errorCode, string errorErrorMessage)
+		{
+			ErrorCode = errorCode;
+			ErrorMessage = string.IsNullOrEmpty(errorErrorMessage)
+				? errorErrorMessage
+				: GetDescription(errorCode);
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ExecutionResult"/> class.
+		/// </summary>
+		protected ExecutionResult(): this(ExecutionErrorCode.None, null)
+		{
+		}
+
+		/// <summary>
+		/// Gets the ErrorCode value which is representing execution result.
+		/// </summary>
+		public ExecutionErrorCode ErrorCode { get; }
+
+		/// <summary>
+		/// Gets the error message related with the execution result.
+		/// </summary>
+		public string ErrorMessage { get; }
+
+		/// <summary>
+		/// Gets a value indicating whether the execution is successful.
+		/// </summary>
+		public bool IsSuccess => ErrorCode == ExecutionErrorCode.None;
+
+
+		/// <summary>
+		/// Performs an implicit conversion from <see cref="ExecutionErrorCode"/> to <see cref="ExecutionResult"/>.
+		/// </summary>
+		/// <param name="errorCode">The error code.</param>
+		/// <returns>
+		/// The result of the conversion.
+		/// </returns>
+		public static implicit operator ExecutionResult(ExecutionErrorCode errorCode)
+		{
+			return Error(errorCode);
+		}
+
+		/// <summary>
+		/// Creates the unsuccessful execution result.
+		/// </summary>
+		/// <param name="errorCode">The error code.</param>
+		/// <param name="errorMessage">The error message.</param>
+		/// <returns>
+		/// A <see cref="ExecutionResult" /> instance with <see cref="ExecutionResult.IsSuccess" /> set to <c>false</c>.
+		/// </returns>
+		public static ExecutionResult Error(ExecutionErrorCode errorCode, string errorMessage = null)
+		{
+			return new ExecutionResult(errorCode, errorMessage);
+		}
+
+		/// <summary>
+		/// Creates the successful execution result.
+		/// </summary>
+		/// <returns>
+		/// A <see cref="ExecutionResult" /> instance with <see cref="ExecutionResult.IsSuccess" /> set to <c>true</c>.
+		/// </returns>
+		public static ExecutionResult Success()
+		{
+			return new ExecutionResult();
+		}
+
+		#region Private Methods
+
+		/// <summary>
+		/// Gets the description for <param name="executionErrorCode"/>.
+		/// </summary>
+		/// <param name="executionErrorCode">The execution error code.</param>
+		/// <returns>
+		/// If enum value marked by <see cref="DescriptionAttribute"/> then the resulted value will be obtained from it.
+		/// In other case result will be a string split by Upper case letters.
+		/// E.g. for InputCommandInvalidArguments result will be 'Input Command Invalid Arguments'
+		/// </returns>
+		private string GetDescription(ExecutionErrorCode executionErrorCode)
+		{
+			FieldInfo fieldInfo = executionErrorCode.GetType().GetField(executionErrorCode.ToString());
```

Comment:
We should extract `executionErrorCode.ToString` to a variable since it's used twice. Also, in such cases I'd suggest using `ToString("G")` to be more explicit about the values we expect to get.

Reply:
fixed